lib/Launcher.js

```diff
@@ -76,6 +78,13 @@ class Launcher {
 
       chromeArguments.push(`--user-data-dir=${options.userDataDir || temporaryUserDataDir}`);
     }
+    if (options.prefs) {
+      console.assert(!options.headless, `Custom preferences can't be used in headless mode`);
```

Comment:
That's a bummer. I didn't know preferences are not supported by headless browser. We try hard not to land API that works only in headful browser.

@skyiea can you please share what particular preferences are you interested in? We might be able to plumb them for you through DevTools protocol so that it works in both headful and headless modes.

Reply:
I was surprised too that it doesn't work in headless browser, given that headless supports custom `userDataDir` and, say, `localStorage`/`cookies`. Might be some of you guys have some insight why `Preferences` file is being ignored in headless?
Personally for me it makes no difference, because I'm using puppeteer to test Chrome extension, which means that I don't have a chance to use headless mode anyway (BTW, any insight why so?). Thus I guess many of the users are already using puppeteer in headful mode.

That's not only about my particular case, I guess. As I've said, there is a lot (https://src.chromium.org/viewvc/chrome/trunk/src/chrome/common/pref_names.cc?view=markup) of additional settings that are available via preferences. For me it allows to set `prefs.download.default_directory` (`setDownloadBehavior ` doesn't work for me, I guess because I'm downloading via `chrome.downloads` API), and in the future I'd like to be able to trigger some experiments (via `prefs.experiments_enabled`).